db/batch/batch_impl.go

```diff
@@ -248,30 +252,38 @@ func (cb *cachedBatch) Delete(namespace string, key []byte, errorFormat string,
 	cb.lock.Lock()
 	defer cb.lock.Unlock()
 	h := cb.hash(namespace, key)
-	cb.Evict(h)
+	cb.currentCache().Evict(h)
```

Comment:
should also do
```
for i := len(cb.caches) - 1; i >= 0; i-- {
   cb.caches[i].Evict(h)
}
```?
key to delete may not be in currentCache

Reply:
can you give an example of this case?

Reply:
```
Put(k1, v1)
Put(k2, v2)
Snapshot()
Put(k3, v3)  // currentCache() = caches[1]
Delete(k1)  // k1 is in caches[0]
```

Reply:
oh, it depends on the `deleted` map in `currentCache` to indicate a key has been removed, the <k, v> still exist in earlier cache (like caches[0]) and cannot be removed (otherwise Revert() will fail)
as long as Snapshot/Revert and when CommitBatch(cb) works correctly, i think should be fine